File: Arcade/Intro/10.clj
(defn commonCharacterCount [s1 s2]
  "Count the numbers of common character between two strings."
  (let [inter (clojure.set/intersection (set s1) (set s2))
        dict1 (frequencies s1)
        dict2 (frequencies s2)]
    (apply +
           (map #(min (dict1 %) (dict2 %)) inter))))

File: Arcade/Intro/10.cpp
int commonCharacterCount(string s1, string s2) {
  // commoncharactercount : Str, Str -> Int.
  // Count the numbers of common character between two strings.
  int count = 0;                // Counter of common chars
  for(auto& c : s1) {
    int pos = s2.find(c);
    if(pos != -1) {
      count += 1;
      s2.erase(pos, 1);
    }
  }
  return count;
}
